modules/programs/vim.nix

```diff
@@ -0,0 +1,79 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.vim;
+  defaultPlugins = [ "sensible" ];
+
+in
+
+{
+  options = {
+    programs.vim = {
+      enable = mkEnableOption "Vim";
+
+      lineNumbers = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "Weather to show line numbers.";
+      };
+
+      tabSize = mkOption {
+        type = types.nullOr types.int;
+        default = null; 
+        example = 4;
+        description = "Set tab size and shift width to a specified number of sapces.";
+      };
+
+      plugins = mkOption {
+        type = types.listOf types.str;
+        default = defaultPlugins;
+        example = literalExample ''
+          extraPlugins: [ "YankRing" ]
+        '';
+        description = ''
+          List of vim plugins to install.
+          For supported plugins see: https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/vim-plugins/vim-plugin-names
+        '';
+      };
+
+      rcExtra = mkOption {
```

Comment:
I think this should be `extraConfig` instead of `rcExtra` to be more consistent with other NixOS/Home Manager options.